server/app.js

```diff
@@ -15,45 +16,170 @@ app.use(express.static(__dirname + "/public" ));
 app.get('/', function(req,res){
     res.redirect('index.html');
 });
- 
-var client = new net.Socket();
+
+require('events').EventEmitter.prototype._maxListeners = 100; //max listeners
+
+var PythonShell = require('python-shell');
+
+var client = new net.Socket();  
+
+client.on('err', function(err){
+            // handle the error safely
+            console.log("Client error");
+            console.log(err)
+        });
 
 io.on('connection',function(socket){ 
+
+    //////////////////////////////////////////////////////////////////
+    // Logging server stuff //
+    var server = net.createServer(function(sock) {
+        sock.on('err', function(err){
+            console.log("vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv\n\n");
+            console.log(err);
+            console.log("\n\n^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^");
+            server.close();
+            server.listen(5002, '192.168.43.12');  
```

Comment:
These hardcoded values should be at the top of the script or somewhere for easier configuration.